script.js

```diff
@@ -1,4 +1,204 @@
+// Track number of players in the game
+var numPlayers = 0;
+
+// Track number of dice to use
+var numDice = 0;
+
+// Track game version
+var gameVersion = "";
+
+// Track game mode - allows the game to move into the different player stages eg. rolling dice, getting sequence etc
+var gameMode = 1;
+
+// Variable to store a player's dice rolls
+var playerRoll = [];
+
+// Variable to store all player's combined numbers
+var playerCombinedNums = [];
+
+// Variable to track running scores of each player.
+var runningScore = [];
+
+// Function to get user inout for game version.
+var getVersion = function (input) {
+  gameVersion = input;
+  return `In this mode, the ${gameVersion} number will win the game. <br><br> Please input the number of players.`;
+};
+
+// Function to generate dice roll
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+//Request for player input to get number of players in game.
+var getPlayers = function (input) {
+  numPlayers = input;
+  return `Welcome! There are ${numPlayers} players in this game. <br><br> Please select how many dice you want to play with. `;
+};
+
+// Get player input to number of dice used in game.
+var getNumDice = function (input) {
+  numDice = input;
+  return `You will be playing with ${numDice} dice in this game. <br><br> Player 1, please click "submit" to roll the dice.`;
+};
+
+// Get player to roll dice.
+var getPlayerRoll = function () {
+  var index = 0;
+  var playerDiceRoll = [];
+  while (index < numDice) {
+    playerDiceRoll.push(diceRoll());
+    index += 1;
+  }
+  return playerDiceRoll;
+};
+
+// Get player to select the sequence of the dice. Function takes in the sequence input, and the array of dice rolls.
+var getSequence = function (input, diceArray) {
+  var combinedNum = 0;
+  var autoCombinedNum = 0;
+  var indexArray = input.split("").map(Number);
+  // This loop takes in the user input sequence and generate the combined number.
```

Comment:
Interesting use of the map method. Could you explain what this line of code (line60) does?

Reply:
haha I took this off the hint in the initial project question, before Kai updated it to remove the "Get user to input sequence of dice" requirement. 

As I understand though, the user input is a string, the split method will then break up each character in the string since the delimiter is "". The map method then applies a function to all elements that have been split up. In this case, it applies the Number function, which converts all elements to an integer data type in the array. 

This is necessary so we can make use of each element in IndexArray to act as an index for the diceArray array in line 63.